control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsHandler.java

```diff
@@ -0,0 +1,108 @@
+package org.oskari.announcements.actions;
+
+import fi.nls.oskari.control.RestActionHandler;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.log.AuditLog;
+import fi.nls.oskari.util.JSONHelper;
+
+@OskariActionRoute("Announcements")
+public class AnnouncementsHandler extends RestActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(AnnouncementsHandler.class);
+    private AnnouncementsService service = new AnnouncementsServiceMybatisImpl();
+
+    // Handle get announcements
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+
+        try {
+            JSONObject result = new JSONObject();
+            if (params.getUser().isAdmin()) {
+                result = service.getAdminAnnouncements();
+            } else {
+                result = service.getAnnouncements();
+            }
+            System.out.println(result);
+            ResponseHelper.writeResponse(params, 200, result);
+        
+        } catch (Exception e) {
+            LOG.error("Error", e);
+        }
+    }
+    // Handle update announcements
+    @Override
+    public void handlePut(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        try {
+            Announcement announcement = AnnouncementsParser.parseAnnouncementParams(params);
+            final JSONObject result = new JSONObject();
+            int updateId = service.updateAnnouncement(announcement);
+            JSONHelper.putValue(result, "id", updateId);
+            
+            AuditLog.user(params.getClientIp(), params.getUser())
+            .withParam("id", announcement.getId())
+            .withParam("title", announcement.getTitle())
+            .withParam("content", announcement.getContent())
+            .withParam("beginDate", announcement.getBeginDate())
+            .withParam("endDate", announcement.getEndDate())
+            .withParam("active", announcement.getActive());
+
+            ResponseHelper.writeResponse(params, result);
+        } catch (JSONException e) {
+            LOG.error("Error updating announcement", e);
```

Comment:
No need to LOG.error() here. Since you are throwing an ActionException it will be logged anyway by the controller. Logging it twice only adds spam to the log files.